scripts/travis/permissions-workaround.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ "$(whoami)" != 'root' ]; then
+    echo 'Please run with sudo'
+    exit 2
+fi
+
+UBUNTU_VERSION=`lsb_release -a 2>&1 | grep Codename | grep -v "LSB" | awk '{print $2}'`
+
+# Workaround for Travis on Ubuntu Xenial so local runner integration tests work
+# when executing them under user "stanley" (by default Travis checks out the
+# code and runs tests under a different system user).
+if [ "${UBUNTU_VERSION}" == "xenial" ]; then
+  echo "Applying workaround for stanley user permissions issue to /home/travis on Xenial"
+  chmod -R a+rwX /home/travis
```

Comment:
`a+rwX` is better than `777` as it was before. We can change this to only manipulate the other permissions: `o+rwX`.
`r` because we're having issues accessing from another user.
`w` because python will want to write pyc files and other things might need temp files as well.
`X` only sets the execute bit when a file is already executable for the user. Primarily that helps with directories (for listing contents) and actually executable files.

We can probably have a more explicit list of directories to recurse into, but I'm not sure what all is used, so we'd have to experiment with that.

I think changing the ownership of files in a home directory is generally a bad idea.

Another thought: What about changing the st2.test.conf to run things as travis instead of as stanley?

Reply:
Awesome, thanks for the explanation there. That's something we should throw in a comment, so when people (or me) run across this in the future they don't have to dig around for why we're setting things they way they are. :smile:

Reply:
#4869 included a simplified permissions fix, so I've removed all the permissions stuff from this PR :) (except for one commit where I removed some --exe flags).